src/components/ErrorBoundary/ErrorBoundary.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * @flow
+ * @file Error Boundary
+ * @author Box
+ */
+
+import React from 'react';
+import noop from 'lodash/noop';
+
+type Props = {
+    children?: any,
+    errorComponent: any,
+    onError: Function,
+    render?: Function,
+};
+
+type State = {
+    error?: Error,
+};
+
+class ErrorBoundary extends React.Component<Props, State> {
+    props: Props;
+    state: State = {};
+
+    static defaultProps = {
+        errorComponent: null,
+        onError: noop,
+    };
+
+    componentDidCatch(error: Error): void {
+        this.setState({ error }, () => {
+            this.props.onError(error);
+        });
+    }
+
+    render() {
+        const { errorComponent, render }: Props = this.props;
+        const { error }: State = this.state;
+
+        if (render) {
+            return render({ error });
+        }
+
```

Comment:
Can you explain these two functions, since I don't see it being passed?

Reply:
A render prop surfaces a component's internal state so it can be used to render arbitrary markup. I added it to make the component more flexible, even though it's not being used at the moment. I can remove it if that's not desirable. 

Example usage:

```
<ErrorBoundary render={(error) => (
    <h1>There was a problem on the page: {error.message}</h1>
)>
   <div>Here's my happy path markup</div>
</ErrorBoundary>
```

Reply:
I would probably just remove it for now till we end up using it that way. It will also make your tests easier due to less branch logic.